### PR TITLE
Test adding matrices

### PR DESCRIPTION
**Pull request by arthbous**
New branch for **aux/test_dof/** and for **tests/add_mat_tests/**

It will test the additions of the matrices but not EAFE

### CODE REVIEW

**Comment by arthbous** [comment]
Right now it works but I have to replace the values of the big matrix by the value of the small matrix. I can not add.

Which technically would work with EAFE, since the EAFE matrix replace the value of the big matrix.

**Comment by maximilianmetti** [comment]
How are you replacing the entries of the matrix?  Ultimately, we might not want to *replace* the PNP blocks, but add on top of them.  I think we should have an EAFE-coefficient multiplying the terms that would be replaced by EAFE, in case there need to be some boundary/surface integrals that shouldn't be removed.  Then addition of the EAFE blocks would be perfect.

**Comment by arthbous** [comment]
I am replacing by doing **vals[kk]=el;** instead of **vals[kk]+=el;**. The problem is as follows:

If we add
* a1   = inner(grad(u1), grad(v1))*dx
* a  = Eps*inner(grad(u1), grad(v1))*dx+inner(grad(u2), grad(v2))*dx

then if you Eps=0, when you assemble a into an EigenMatrix A. A will not have any entries for the DoF u1 x v1. That's a big problem. So 4 solutions:

* You take Eps = 100 and you replace the values of the matrix. In this case it works there are no overlap of DOF
* You take Eps=1E-20, and then you can add because the matrix will have some entries. It consider 1E-22 to be non zero in the sparse storage.
* We find a way to understand their horrible documentation about A.add()
* I can code our own version of A.add()

**Comment by maximilianmetti** [comment]
Another solution, replace with EAFE as you currently do. If there need to be extra terms from a surface integral, which we can deal with later, we can do an addition. This type of situation can only arise when using Robin boundary conditions by the way.

**Comment by arthbous** [comment]
So we will do two function, one replace and one add

**Comment by maximilianmetti** [comment]
We can start with just replacement.

**Comment by arthbous** [comment]
It's all done. You should test it and merge it.

Run the following:
* **aux_test_dof**
* **test_add_matrix**
* **test_replace_matrix**

And you can take a look at src/funcspace_to_vecspace.cpp.

File: tests/matrices_tests/test_add_matrix.cpp [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+  Constant u2(0.0,0.0);
+  DirichletBC bc2(V2, u2, boundary);
+  VecSpace2::BilinearForm a2(V2, V2);
+  VecSpace2::LinearForm L2(V2);
+  L2.f1 = f;
+  L2.f2 = f;
+  a2.Ep1 = Ep1;
+  a2.Ep2 = Ep1;
+  L2.Ep1 = Ep2;
+  L2.Ep2 = Ep2;
```
Comment (maximilianmetti) [inline tests/matrices_tests/test_add_matrix.cpp]: It looks like these assignments got messed up...

**Comment by maximilianmetti** [comment]
Seems to work!

**Pull request merged**